module/VuFind/src/VuFind/RecordDriver/SolrDefault.php

```diff
@@ -234,11 +235,15 @@ public function getHighlightedSnippet()
         if ($this->snippet) {
             // First check for preferred fields:
             foreach ($this->preferredSnippetFields as $current) {
-                if (isset($this->highlightDetails[$current][0])) {
-                    return [
-                        'snippet' => $this->highlightDetails[$current][0],
-                        'caption' => $this->getSnippetCaption($current),
-                    ];
+                if (array_key_exists($current, $this->highlightDetails)) {
+                    foreach ($this->highlightDetails[$current] as $hl) {
```

Comment:
A small suggested simplification -- instead of the `array_key_exists check` (which adds a whole extra layer of indentation), just do:

`foreach ($this->highlightDetails[$current] ?? [] as $hl) {`